examples/DS3231_Temperature/DS3231_Temperature.ino

```diff
@@ -0,0 +1,164 @@
+    // the available pins for SDA, SCL
+    // Wire.begin(0, 2); // due to limited pins, use pin 0 and 2 for SDA, SCL
+    
+    Rtc.Begin();
+
+    RtcDateTime compiled = RtcDateTime(__DATE__, __TIME__);
+    printDateTime(compiled);
+    Serial.println();
+
+    if (!Rtc.IsDateTimeValid()) 
+    {
+        // Common Causes:
+        //    1) first time you ran and the device wasn't running yet
+        //    2) the battery on the device is low or even missing
+
+        Serial.println("RTC lost confidence in the DateTime!");
+
+        // following line sets the RTC to the date & time this sketch was compiled
+        // it will also reset the valid flag internally unless the Rtc device is
+        // having an issue
+
+        Rtc.SetDateTime(compiled);
+    }
+
+    if (!Rtc.GetIsRunning())
+    {
+        Serial.println("RTC was not actively running, starting now");
+        Rtc.SetIsRunning(true);
+    }
+
+    RtcDateTime now = Rtc.GetDateTime();
+    if (now < compiled) 
+    {
+        Serial.println("RTC is older than compile time!  (Updating DateTime)");
+        Rtc.SetDateTime(compiled);
+    }
+    else if (now > compiled) 
+    {
+        Serial.println("RTC is newer than compile time. (this is expected)");
+    }
+    else if (now == compiled) 
+    {
+        Serial.println("RTC is the same as compile time! (not expected but all is fine)");
+    }
+    Serial.println();
+
+    // never assume the Rtc was last configured by you, so
+    // just clear them to your needed state
+    Rtc.Enable32kHzPin(false);
+    Rtc.SetSquareWavePin(DS3231SquareWavePin_ModeNone);
+    
+} //setup
+
+// Temperature test loop
+void loop () 
+{
+    if (!Rtc.IsDateTimeValid()) 
+    {
+        // Common Causes:
+        //    1) the battery on the device is low or even missing and the power line was disconnected
+        Serial.println("RTC lost confidence in the DateTime!");
+    }
+
+    // Force a temperature A/D conversion
+    Rtc.ForceTemperatureCompensationUpdate( true );
+
+    // Output temperature in various formats
+    printTemps();    
+
+    delay(100);  // Fast enough to watch quick temperature drifts
+        
+} //loop
+
+#define countof(a) (sizeof(a) / sizeof(a[0]))
+
+void printDateTime(const RtcDateTime& dt)
+{
+    char datestring[20];
+
+    snprintf_P(datestring, 
+            countof(datestring),
+            PSTR("%02u/%02u/%04u %02u:%02u:%02u"),
+            dt.Month(),
+            dt.Day(),
+            dt.Year(),
+            dt.Hour(),
+            dt.Minute(),
+            dt.Second() );
+    Serial.print(datestring);
+    
+} //printDateTime
+
+void printTemps( void )
+{
+    // Get the temperature data    
+    RtcTemperature temp = Rtc.GetTemperature();   
+
+    // Print:
+    // a) Floating point temperature
+    //    Selectively add spaces to create const. width field
+    //    ( no float formating for 'printf' type functions )
+
+    float  tf = temp.AsFloat();
```

Comment:
separate the print float and print fractional into their own functions, makes this easier to read for beginners.

Reply:
Have separated them out using 'Serial.printf()' statements. A new method, 'IsMinusSignNeeded()', has been added to get around a conceptual error regarding 'AsWholeDegrees()' and 'GetFractional()'. See pull request (#51) for more details.